homeassistant/components/maxcube/climate.py

```diff
@@ -95,13 +109,72 @@ def current_temperature(self):
 
     @property
     def hvac_mode(self):
-        """Return current operation (auto, manual, boost, vacation)."""
+        """Return current operation mode."""
+        device = self._cubehandle.cube.device_by_rf(self._rf_address)
+        if device.mode == MAX_DEVICE_MODE_MANUAL:
+            if device.target_temperature == OFF_TEMPERATURE:
+                return HVAC_MODE_OFF
+            return HVAC_MODE_HEAT
         return HVAC_MODE_AUTO
```

Comment:
The behavior of MAX! Cube in automatic / scheduled mode (`MAX_DEVICE_MODE_AUTOMATIC`) is that it still allows user to change target temperature, but it will be temporary. The temperature will be reset to scheduled one as soon as next setpoint time comes or if you open and close the window. To preserve user set temperature, MAX! Cube has to be set into manual operation mode, which I mapped to Home Assistant's `HVAC_MODE_HEAT`. If I'm not mistaken, vacation mode (`MAX_DEVICE_MODE_VACATION`) behaves the same as automatic mode. However, there is only one setpoint time - it exits vacation mode and switches to automatic one.

Reply:
Eg Google assistant will not allow changing the temperature in our `HVAC_MODE_AUTO`. I'm not sure if there's a better mapping. For greater flexibility in the modes we recommend using the preset feature since the integration can define presets itself. Maybe extend the presets instead?

Reply:
I don't think that MAX! Cube's automatic mode intended temperature customization. Because any changes will be reset by the Cube as soon as some event happens (like schedule-relevant time change, window open / close). It's just that they still allow you to do that. For me this behaviour is much closer to `HVAC_MODE_AUTO` than to `HVAC_MODE_HEAT`. If someone wants to change temperature to something different from currently scheduled one, they would most certainly want to preserve it, so they would have to change to manual mode anyway.

So I would strongly prefer leaving mapping of `MAX_DEVICE_MODE_AUTOMATIC` to `HVAC_MODE_AUTO`.

Vacation mode is intended to be used when you leave for longer periods of time, so that heating returns to automatic mode at some predefined date (e.g., the date when you return back). There's no-one at home to change the temperature while you're on vacation.

`MAX_DEVICE_MODE_VACATION` is already mapped to `PRESET_AWAY`. The only question is which HVAC mode sounds more logical for this preset - `HVAC_MODE_HEAT` or `HVAC_MODE_AUTO`?

Reply:
Can the user set the temperature in vacation mode?

Reply:
You can select temperature when you activate vacation mode. You should also set a date when vacation mode should end (which doesn't seem to be supported by HA). I just tried activating vacation mode and changing temperature directly on the thermostat - it doesn't let me. It even ignores opening / closing of windows. However, I can change temperature through their official MAX! App. Again, by the behavior of the thermostat, it looks like it's not supposed to be changed, but API still allows it.

Reply:
If the temperature will be fixed during the vacation mode, it sounds best to set the home assistant mode to `HVAC_MODE_HEAT` when selecting the away preset.

Reply:
Yeah, it's supposed to be fixed up until vacation mode expires. So it's `HVAC_MODE_HEAT`, then.

Reply:
Good!